### ISSUE TITLE
Unlike version 3.2 the language cookie is ignored for URLs without language code in 3.4

### ISSUE THREAD

**Issue by chrisxchrisxchrisx** [joomla/joomla-cms#7434]
I thought this had been fixed as per https://github.com/joomla/joomla-cms/pull/6452 but the following demonstrates a useful feature which worked in 3.2 but fails in 3.4

1. Using Firefox 39.0 clear all cookies

2. Check Language Filter settings via http://saatalk.info/administrator
Language Selection for new Visitors: Browser Settings
Automatic Language Change: Yes
Item Associations: Yes
Remove URL Language Code: Yes
Cookie Lifetime: Year
Add Alternate Meta Tags: Yes

3. Simulate new visit via http://saatalk.info
Correctly displays British English and stores en-GB cookie in line with browser settings

4. Select Deutsch via language switcher to http://saatalk.info/de
Correctly displays page in German and stores de-DE cookie in line with URL

4. Check http://saatalk.info/meetings
Version 3.2 correctly displays page in German in line with previously stored cookie
Version 3.4 INCORRECTLY displays page in English and stores en-GB cookie in line with Browser Settings but ignoring previously stored cookie

What I want to happen is for the language to be specified by the following in order of priority:
1. via the URL if language code specified - and set the language cookie accordingly
2. else via the language cookie if previously set
3. else via the Browser Settings

Which is what my previous version, 3.2 did

In my test https://github.com/joomla/joomla-cms/pull/7427 failed to fix this problem.

**Comment by zero-24** [comment]
@chrisxchrisxchrisx I think some text is missing as you have only one point / step (1)?

**Comment by Bakual** [comment]
Can you test if https://github.com/joomla/joomla-cms/pull/7427 solves your issue?
You can use the Patchtester component to test the PR: https://docs.joomla.org/Component_Patchtester_for_Testers

**Comment by chrisxchrisxchrisx** [comment]
Thanks Bakual you will have guessed I am new to this forum so will figure out how to test #7427

**Comment by chrisxchrisxchrisx** [comment]
Thank you Bakual for your help. I installed Patchtester locally and used it to apply #7427 and unfortunately it does not fix this problem

**Comment by Bakual** [comment]
Can you add that comment also on the PR? That would help.

**Comment by chrisxchrisxchrisx** [comment]
Thank you Bakual for your patience in guiding me through the process, I'm afraid I rather jumped in without learning the ropes...

**Comment by Bakual** [comment]
You're welcome. It's no problem!

**Comment by infograf768** [comment]
You are using Falang. Please contact the developer.

**Comment by chrisxchrisxchrisx** [comment]
@infograf768 That is a good point, thank you. I am sorry to have wasted your time guys. I would love to help test the native Joomla Language tools but am short of time and expertise right now, even to get it working properly. I found that following the Joomla upgrade I had to make ensure that the Language FIlter settings did not remove the URL language code for Falang to work as I wanted, so that at least links reverted to the browser settings for URLs with no language code.
